### PR TITLE
How P3 Works

### PR DESCRIPTION
**Pull request by diana-macrometa**
Add info about P3 architecture.

### CODE REVIEW

File: docs/photoniq/p3/how-p3-works.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+sidebar_position: 10
+title: How P3 Works
+---
+
+PhotonIQ Performance Proxy (P3) revolutionizes web performance optimization by intercepting and enhancing web requests through a sophisticated, server-side processing environment located at the network edge. This innovative approach enables dynamic optimization of web content, from HTML to multimedia, ensuring faster page loads and improved user experience across different devices and networks.
```
Comment (dwivedianurag) [inline docs/photoniq/p3/how-p3-works.md]: currently we are not doing anything with images and anything .. so please reconsider adding multimedia here

File: docs/photoniq/p3/how-p3-works.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+By centralizing the application of optimization rules, P3 offloads much of the resource-intensive tasks from the end user’s browser to a more capable, server-side platform. This shift not only accelerates content delivery but also maintains high fidelity of the original web design by intelligently applying optimizations without altering the fundamental user experience. Additionally, P3’s architecture is designed to seamlessly integrate with existing web infrastructures, making it a versatile solution for websites aiming to enhance performance without extensive overhaul.
```
Comment (dwivedianurag) [inline docs/photoniq/p3/how-p3-works.md]: is this chatgpt cause we should have here which was earlier

File: docs/photoniq/p3/how-p3-works.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+Here’s a high-level breakdown of how the system processes a request:
+
+1. **Initial Request Handling**:
+   - A user accesses a URL through a desktop or mobile browser, sending a request to the Content Delivery Network (CDN).
+   - The CDN, configured with specific URLs for P3 optimization, redirects eligible requests to the P3 system.
+
+2. **Request Routing to P3**:
+   - Requests are first received by the PhotonIQ API Gateway.
+   - The API Gateway forwards the request to the P3 proxy pod.
```
Comment (dwivedianurag) [inline docs/photoniq/p3/how-p3-works.md]: remove proxy pod

File: docs/photoniq/p3/how-p3-works.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+   - If the HTML ReWriter lacks the rules for a request, then it sends a query to the Controller.
+   - The Controller checks configured policies to see if the URL matches any existing patterns.
+   - If no matching policies are found, then the request is processed without P3 optimizations and passed through as is from the origin.
+
+5. **Job and Policy Handling**:
+   - When a matching policy exists but no current job covers the request, the Controller initiates a new job.
+   - This job is handled by the Analyzer and Validator to ensure the new rules are effective and do not break page functionality or aesthetics.
+
+6. **Optimization Rule Validation**:
+   - The Analyzer assesses the page against the configured optimizers and formulates a set of rules.
```
Comment (dwivedianurag) [inline docs/photoniq/p3/how-p3-works.md]: is better to mention something about machine learning here

Comment (diana-macrometa) [reply docs/photoniq/p3/how-p3-works.md]: Like what? "The analyzer uses machine learning to assess the page..."?

File: docs/photoniq/p3/how-p3-works.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+## P3 Control Flows Diagram
+
+The control flow diagram illustrates the detailed interactions within the P3 architecture.
+
+![P3 Control Flows](/img/photoniq/p3/p3-control-flows.png)
+
+- The process begins when a request reaches the Controller, which determines the necessity for creating a job based on the incoming URL and associated policies.
+- Components subscribe to updates from this job, processing the request as outlined:
+  - **Analyzer**: Takes the job details, including policy ID and URL pattern, and determines safe optimizations.
```
Comment (dwivedianurag) [inline docs/photoniq/p3/how-p3-works.md]: also to add other details as well

Comment (diana-macrometa) [reply docs/photoniq/p3/how-p3-works.md]: Like what details?

File: docs/photoniq/p3/index.md [hunk 6]
```diff
@@ -9,6 +9,12 @@ As a fully-managed service, P3 greatly reduces the need for ongoing technical ov
 
 Enhancing core web vitals, P3 improves user engagement and search engine visibility, which is pivotal in driving organic traffic and elevating revenue opportunities. With P3, technical teams can confidently future-proof their websites, achieving enduring performance enhancements with streamlined effort.
 
+## How It Works
+
+P3 operates based on a comprehensive offline analysis of web pages and their associated resources. This allows for the dynamic updating and caching of content to achieve optimal performance.
```
Comment (dwivedianurag) [inline docs/photoniq/p3/index.md]: it looks good but if we can reword it then it will be good

**Review by dwivedianurag — APPROVED** [review]
ok please review the comment

File: docs/photoniq/p3/index.md (hunk 6 above)
Comment (diana-macrometa) [reply docs/photoniq/p3/index.md]: This is the old architecture description. What don't you like about it?

File: docs/photoniq/p3/how-p3-works.md (hunk 5 above)
Comment (dwivedianurag) [reply docs/photoniq/p3/how-p3-works.md]: something like this --->. `and other policy details which the analyzer will need in order to analyze`

File: docs/photoniq/p3/index.md (hunk 6 above)
Comment (dwivedianurag) [reply docs/photoniq/p3/index.md]: @diana-macrometa we came up with some documentation for P3 and we can leave it for now like this

**Pull request merged**